### PR TITLE
Migrate all packages to unstable

### PR DESCRIPTION
**Pull request by thiagokokada**
# Context

The idea of this PR is to migrate all packages to a pinned commit in `nixpkgs-unstable`. This brings some advantages:

- We ensure that every Nubanker uses the same derivations, including its dependencies
- Should avoid issues between multiple NixOS versions (i.e.: old stable vs. stable vs. unstable)
- Make sure our tooling is up-to-date

Of course, there is some disadvantages too:

- We either pin a commit and needs to bump it from time to time, or  track master and possible lose control and risk some regressions

For now I think pinning should be ok. We could make a CI that would bump the commit from time-to-time and asks for a review. Of course, this is for a future PR.

BTW, since this commit uses unstable for everything, and `clojure-lsp` is currently broken in unstable, this also backports a fix from @ericdallo fork (until it is merged in upstream).

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,70 +1,88 @@
...
       # ruby
-      (yarn.override ({ nodejs = nodejsNubank; }))
     ];
 
-    clojure-tools = with pkgs; [
-      # TODO: graalvm build is failing on master and this triggers a long build
-      # clj-kondo
+    clojure-tools = with unstable; [
+      babashka
+      clj-kondo
```
Comment (ericdallo) [inline default.nix:71]: I think we should add in another PR: `cljfmt`

**Review by ericdallo — APPROVED** [review]
Nice, I like this way

File: default.nix [hunk 2]
```diff
@@ -1,70 +1,88 @@
...
     flutter-patch = callPackage ./pkgs/flutter-patch {};
 
+    nodejs = unstable.nodejs-10_x;
+
+    yarn = (unstable.yarn.override {
+      nodejs = nodejs;
+    });
+
+    leiningen = (unstable.leiningen.override {
+      jdk = unstable.openjdk8;
```
Comment (ericdallo) [inline default.nix:36]: Are we still using jdk8?

Comment (thiagokokada) [reply default.nix:36]: I think there is still some problematic repositories no :thinking: ?

Comment (thiagokokada) [reply default.nix:36]: If you want we can change to JDK11 and see what it breaks :man_shrugging: .

File: default.nix (hunk 1 above)
Comment (thiagokokada) [reply default.nix:71]: Didn't find this in nixpkgs. Do you know the name of the package :thinking: ?

**Comment by thiagokokada** [comment]
Going to merge, future issues can be addressed in other PRs.

**Pull request merged**